docs/rest-api.yml

```diff
@@ -435,21 +447,70 @@ paths:
           description: The user identified by the cookie
           schema:
             $ref: '#/definitions/User'
-  /shutdown:
+        '404':
+          description: A user is not found.
+  /oath2/authorize:
+    get:
+      summary: 'OAuth 2.0 authorize endpoint'
+      description: 'OAuth 2.0 authorize endpoint'
+      parameters:
+        - name: client_id
+          description: The client id
+          in: path
+          required: true
+          type: string
+        - name: response_type
+          description: The response type (token | code)
+          in: path
+          required: true
+          type: string
+        - name: state
+          description: A state string
+          in: path
+          required: false
+          type: string
+        - name: redirect_uri
+          description: The redirect url
+          in: path
+          required: false
+          type: string
+        - name: scope
+          description: The requested scopes
+          in: path
+          required: false
+          type: string
+      responses:
+  /oath2/token:
     post:
-      summary: Shutdown the Hub
+      summary: Request an OAuth2 token
+      description: Request an OAuth2 token
       parameters:
-        - name: proxy
-          in: body
-          type: boolean
-          description: Whether the proxy should be shutdown as well (default from Hub config)
-        - name: servers
-          in: body
-          type: boolean
-          description: Whether users's notebook servers should be shutdown as well (default from Hub config)
+        - name: client_id
```

Comment:
/oauth2/token takes the following arguments:

- client_id
- client_secret
- grant_type (always 'authorization_code' for now)
- code
- redirect_url

All arguments are strings and required and `in: body`

Response is JSON, of the form:

```python
{ 'access_token': token}
```